dipy/reconst/tests/test_cti.py

```diff
@@ -0,0 +1,421 @@
+        assert np.allclose(
+            cti_pred_signals, qti_pred_signals), "CTI and QTI signals do not match!"
+
+
+def test_split_cti_param():
+    ctiM = cti.CorrelationTensorModel(gtab1, gtab2) #fit_method is WLS by default.
+    for DTD in DTDs: #generating cti_pred_signals for all DTDs
+        D = np.mean(DTD, axis=0)
+        evals, evecs = decompose_tensor(D)
+        C = qti.dtd_covariance(DTD)
+        C = qti.from_6x6_to_21x1(C)
+
+        # getting C_params from voigt notation
+        ccti = modify_C_params(C)
+
+        MD = mean_diffusivity(evals)  # is a sclar
+        # Compute kurtosis tensor (K)
+        K = generate_K(ccti, MD)
+
+        cti_params = construct_cti_params(evals, evecs, K, ccti)
+        # Generate predicted signals using cti_prediction function
+        ctiM = cti.CorrelationTensorModel(gtab1, gtab2) 
+        cti_pred_signals = ctiM.predict(cti_params)
+        #DWI = np.zeros((2, 2, 1, len(gtab.bvals)))
+        #DWI[0, 0, 0] = DWI[0, 1, 0] = DWI[1, 0, 0] = DWI[1, 1, 0] = cti_pred_signals
+        ctiF = ctiM.fit(cti_pred_signals) #should pass cti_pred_signal, in our case we don't have DWI
+        # ctiF = ctiM.fit(cti_params)
+        evals, evecs, kt, cvt = cti.split_cti_params(ctiF.model_params)
+        print('this is ctiF.kt: and its type: ', ctiF.kt, type(ctiF.kt))
+        assert_array_almost_equal(evals, ctiF.evals)
+        assert_array_almost_equal(evecs, ctiF.evecs)
+        assert_array_almost_equal(kt, ctiF.kt)
+        assert_array_almost_equal(cvt, ctiF.cvt)
+
+
+def test_cti_fits():
+    ctiM = cti.CorrelationTensorModel(gtab1, gtab2)
+    for DTD in DTDs:  # trying out all fits for each DTD.
+        D = np.mean(DTD, axis=0)
+        evals, evecs = decompose_tensor(D)
+        C = qti.dtd_covariance(DTD)
+        C = qti.from_6x6_to_21x1(C)
+
+        # getting C_params from voigt notation
+        ccti = modify_C_params(C)
+
+        MD = mean_diffusivity(evals)  # is a sclar
+        # Compute kurtosis tensor (K)
+        K = generate_K(ccti, MD)
+
+        cti_params = construct_cti_params(evals, evecs, K, ccti)
+        # Generate predicted signals using cti_prediction function
+        cti_pred_signals = ctiM.predict(cti_params)
+
+        # def ls_fit_cti(design_matrix, data, inverse_design_matrix, weights=True,  # shouldn't the effect of covariance tensor be obsvd ?
+        #        min_diffusivity=0):
+        inverse_design_matrix = np.linalg.pinv(design_matrix(gtab1, gtab2))
+        cti_return = ls_fit_cti(design_matrix(gtab1, gtab2), cti_pred_signals, inverse_design_matrix )
+        # OLS fitting
+        ctiM = cti.CorrelationTensorModel(gtab1, gtab2, fit_method="OLS")
+        # ctiF = ctiM.fit(cti_pred_signals)
+        # ctiF = ctiM.fit(cti_params) #this is turning out to be NoneType                     #error 
+
+        # assert_array_almost_equal(ctiF.model_params, cti_params)
+        assert_array_almost_equal(cti_return, cti_params)
```

Comment:
This is failing but it is not an issue of the 'ls_fit_cti' function. 

'cti_return' does not match 'cti_params' because of the eigenvalue decomposition. 

Since the total diffusion tensors of your tests are isotropic, when the tensor are decomposed inside function 'ls_fit_cti' it creates a different eigenvector due to float precision (note that away set of eigenvectors will create the same isotropic diffusion tensor if the tree eigenvalues are equal which seem to be the case). 

If you assert the evals, kurtosis tensor elements and covariance tensor elements, you will see that everything correctly matches. So to fix this issue just replace this line of code by:

    evals, evecs, kt, cvt = split_cti_params(cti_params)
    evals_return, evecs_return, kt_return, cvt_return = split_cti_params(cti_return)
    assert np.allclose(evals, evals_return), "evals do not match!"
    assert np.allclose(kt, kt_return), "K do not match!"
    assert np.allclose(cvt, cvt_return), "C do not match!"